src/main/java/com/google/step/snippet/data/Card.java

```diff
@@ -28,4 +28,28 @@ public String getLink() {
   public String getDescription() {
     return description;
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) {
+      return false;
+    }
+    if (!Card.class.isAssignableFrom(obj.getClass())) {
+      return false;
+    }
+    final Card other = (Card) obj;
+    if (!this.title.equals(other.title)) {
+      return false;
+    }
+    if (!this.code.equals(other.code)) {
+      return false;
+    }
+    if (!this.link.equals(other.link)) {
+      return false;
+    }
+    if (!this.description.equals(other.description)) {
+      return false;
+    }
+    return true;
```

Comment:
For these checks, I think you can collapse it into a single return statement. If one check fails, the other checks won't happen anyways so the single return is equivalent to the list of if checks:
```
return this.title.equals(other.title)
     && this.code.equals(other.code)
     && this.link.equals(other.link)
     && this.description.equals(other.description);
```